File: lib/models/printer_text.dart
class PrinterText {
  final String text;
  // 0 = left / 1 = center / 2 = right / default = 0
  final int textAlign;
  // default 28
  final int textSize;
  // NORMAL = 0 BOLD = 1 ITALIC = 2 BOLD_ITALIC = 3
  final int textStyle;
  // default 1.0f
  final double lineSpacing;

  PrinterText({
    required this.text,
    this.textAlign = 0,
    this.textSize = 28,
    this.textStyle = 0,
    this.lineSpacing = 1.0,
  });

  factory PrinterText.fromJson(Map<String, dynamic> json) {
    return PrinterText(
      text: json['text'] as String,
      textAlign: json['textAlign'] as int,
      textSize: json['textSize'] as int,
      textStyle: json['textStyle'] as int,
      lineSpacing: json['lineSpacing'] as double,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'text': text,
      'textAlign': textAlign,
      'textSize': textSize,
      'textStyle': textStyle,
      'lineSpacing': lineSpacing,
    };
  }
}

File: android/src/main/kotlin/com/ezyorder/flutter_imin_printer/models/PrinterText.kt
package com.ezyorder.flutter_imin_printer.models


import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.squareup.moshi.Moshi
import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory

@JsonClass(generateAdapter = true)
data class PrinterText(
    val text: String,
    @Json(name = "textAlign") val textAlign: Int = 1,
    @Json(name = "textSize") val textSize: Int = 28,
    @Json(name = "textStyle") val textStyle: Int = 0,
    @Json(name = "lineSpacing") val lineSpacing: Float = 1.0f
) {

    companion object {
        fun fromJson(json: String): PrinterText? {
            val moshi = Moshi.Builder()
                .addLast(KotlinJsonAdapterFactory())
                .build()

            val adapter = moshi.adapter(PrinterText::class.java)
            return adapter.fromJson(json)
        }
    }
}

